mlflow/pipelines/steps/evaluate.py

```diff
@@ -197,12 +197,17 @@ def my_warn(*args, **kwargs):
                             "explainability_algorithm": "kernel",
                             "explainability_nsamples": 10,
                             "pos_label": self.positive_class,
+                            "metric_prefix": "validation_",
```

Comment:
Both train and evaluate steps call `mlflow.evaluate` on the validation DF.

Reply:
We definitely shouldn't :)

Reply:
Printed out the validation metrics in train and evaluate steps:

```python
# === train ===
{
    "validation_score": 0.87041291908791,
    "validation_example_count": 959,
    "validation_mean_absolute_error": 1.7805488982087214,
    "validation_mean_squared_error": 15.399400989287159,
    "validation_root_mean_squared_error": 3.924207052295681,
    "validation_sum_on_target": 12612.5,
    "validation_mean_on_target": 13.151720542231491,
    "validation_r2_score": 0.87041291908791,
    "validation_max_error": 52.77168516496291,
    "validation_mean_absolute_percentage_error": 0.15852484851545784,
    "validation_weighted_mean_squared_error": 17.692126139931382,
}


# === evaluate ===
{
    "validation_score": 0.87041291908791,
    "validation_example_count": 959,
    "validation_mean_absolute_error": 1.7805488982087216,
    "validation_mean_squared_error": 15.399400989287159,
    "validation_root_mean_squared_error": 3.924207052295681,
    "validation_sum_on_target": 12612.5,
    "validation_mean_on_target": 13.151720542231491,
    "validation_r2_score": 0.87041291908791,
    "validation_max_error": 52.77168516496291,
    "validation_mean_absolute_percentage_error": 0.15852484851545784,
    "validation_weighted_mean_squared_error": 17.692126139931382,
}
```

They look identical.

Reply:
Awesome! Let's remove the eval step one.

Reply:
https://github.com/harupy/mlflow/blob/d5bc213016b3322e9b68c716e4cfb874fb621fd3/mlflow/pipelines/steps/evaluate.py#L199-L198

Reply:
That's correct. In `evaluate` step we run SHAP but not in `train` step. I'd suggest to keep evaluate over validation dataset in both steps.